colossalai/inference/tensor_parallel/engine.py

```diff
@@ -75,8 +81,12 @@ def _init_manager(self) -> None:
         assert self.tp_size >= 1, "TP size not initialized without providing a valid ShardConfig"
         assert self.head_num % self.tp_size == 0, f"Cannot shard {self.head_num} heads with tp size {self.tp_size}"
         self.head_num //= self.tp_size    # update sharded number of heads
-        self.cache_manager = MemoryManager(self.max_total_token_num, self.dtype, self.head_num, self.head_dim,
-                                           self.layer_num)
+        if self.multi_query_group_num:
```

Comment:
multi_query_group_num should divide tp_size.

Reply:
done